astropy/wcs/wcsapi/low_level_api.py

```diff
@@ -109,7 +115,7 @@ def world_to_pixel_values(self, *world_arrays):
         """
 
     @abc.abstractmethod
-    def world_to_array_index_values(self, *world_arrays):
+    def world_to_array_index_values(self, *world_arrays: scalar_or_array) -> scalar_or_array_or_tuple:
```

Comment:
Could this technically return an int scalar or are we disallowing this?

Reply:
From [PEP 484](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower):

> when an argument is annotated as having type float, an argument of type int is acceptable; similar, for an argument annotated as having type complex, arguments of type float or int are acceptable.

Reply:
Ok perfect